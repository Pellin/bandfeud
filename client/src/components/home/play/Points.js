import React from 'react';
import { motion } from 'framer-motion';

const Points = ({ points }) => {
  return (
    <>
      {points && (
        <motion.div
          className="points"
          initial={{ scale: 0, opacity: 1 }}
          animate={{ scale: 3, opacity: [null, 1, 0] }}
          transition={{ duration: 2, times: [0, 0.7, 1] }}
        >
          +{points}
        </motion.div>
      )}
    </>
  );
};

export default Points;
