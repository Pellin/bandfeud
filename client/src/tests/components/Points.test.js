import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import Points from '../../components/home/play/Points';

it('should render Points correctly', () => {
  const wrapper = mount(<Points points={23} />);
  expect(wrapper.find('div').text()).toBe('+23');
  expect(wrapper).toMatchSnapshot();
});