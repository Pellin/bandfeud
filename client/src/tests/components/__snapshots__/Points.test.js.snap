// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Points correctly 1`] = `
<Points
  points={23}
>
  <ForwardRef(MotionComponent)
    animate={
      Object {
        "opacity": Array [
          null,
          1,
          0,
        ],
        "scale": 4.5,
      }
    }
    className="points"
    initial={
      Object {
        "opacity": 1,
        "scale": 0,
      }
    }
    transition={
      Object {
        "duration": 2,
        "times": Array [
          0,
          0.7,
          1,
        ],
      }
    }
  >
    <ForwardRef(MountRefComponent)
      externalRef={null}
      isStatic={false}
      values={
        MotionValuesMap {
          "hasMounted": true,
          "onUpdate": undefined,
          "output": [Function],
          "unsubscribers": Map {
            "scale" => [Function],
            "opacity" => [Function],
          },
          "values": Map {
            "scale" => MotionValue {
              "canTrackVelocity": true,
              "current": 0,
              "lastUpdated": 0,
              "notifySubscriber": [Function],
              "parent": undefined,
              "prev": undefined,
              "renderSubscribers": Set {
                [Function],
              },
              "scheduleVelocityCheck": [Function],
              "setChild": [Function],
              "stopAnimation": [Function],
              "timeDelta": 0,
              "transformer": undefined,
              "updateAndNotify": [Function],
              "updateSubscribers": Set {
                [Function],
              },
              "velocityCheck": [Function],
            },
            "opacity" => MotionValue {
              "canTrackVelocity": true,
              "current": 1,
              "lastUpdated": 0,
              "notifySubscriber": [Function],
              "parent": undefined,
              "prev": undefined,
              "renderSubscribers": Set {
                [Function],
              },
              "scheduleVelocityCheck": [Function],
              "setChild": [Function],
              "stopAnimation": [Function],
              "timeDelta": 0,
              "transformer": undefined,
              "updateAndNotify": [Function],
              "updateSubscribers": Set {
                [Function],
              },
              "velocityCheck": [Function],
            },
          },
        }
      }
    />
    <Component
      animate={
        Object {
          "opacity": Array [
            null,
            1,
            0,
          ],
          "scale": 4.5,
        }
      }
      controls={
        ValueAnimationControls {
          "activeOverrides": Set {},
          "baseTarget": Object {
            "opacity": 0,
            "scale": 4.5,
          },
          "defaultTransition": Object {
            "duration": 2,
            "times": Array [
              0,
              0.7,
              1,
            ],
          },
          "hasValue": [Function],
          "isAnimating": Set {
            "scale",
            "opacity",
          },
          "makeTargetAnimatable": [Function],
          "overrides": Array [],
          "props": Object {
            "animate": Object {
              "opacity": Array [
                null,
                1,
                0,
              ],
              "scale": 4.5,
            },
            "children": Array [
              "+",
              23,
            ],
            "className": "points",
            "initial": Object {
              "opacity": 1,
              "scale": 0,
            },
            "transition": Object {
              "duration": 2,
              "times": Array [
                0,
                0.7,
                1,
              ],
            },
          },
          "readValueFromSource": [Function],
          "resolvedOverrides": Array [],
          "values": MotionValuesMap {
            "hasMounted": true,
            "onUpdate": undefined,
            "output": [Function],
            "unsubscribers": Map {
              "scale" => [Function],
              "opacity" => [Function],
            },
            "values": Map {
              "scale" => MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "notifySubscriber": [Function],
                "parent": undefined,
                "prev": undefined,
                "renderSubscribers": Set {
                  [Function],
                },
                "scheduleVelocityCheck": [Function],
                "setChild": [Function],
                "stopAnimation": [Function],
                "timeDelta": 0,
                "transformer": undefined,
                "updateAndNotify": [Function],
                "updateSubscribers": Set {
                  [Function],
                },
                "velocityCheck": [Function],
              },
              "opacity" => MotionValue {
                "canTrackVelocity": true,
                "current": 1,
                "lastUpdated": 0,
                "notifySubscriber": [Function],
                "parent": undefined,
                "prev": undefined,
                "renderSubscribers": Set {
                  [Function],
                },
                "scheduleVelocityCheck": [Function],
                "setChild": [Function],
                "stopAnimation": [Function],
                "timeDelta": 0,
                "transformer": undefined,
                "updateAndNotify": [Function],
                "updateSubscribers": Set {
                  [Function],
                },
                "velocityCheck": [Function],
              },
            },
          },
          "variants": Object {},
        }
      }
      inherit={false}
      initial={
        Object {
          "opacity": 1,
          "scale": 0,
        }
      }
      key="animation"
      transition={
        Object {
          "duration": 2,
          "times": Array [
            0,
            0.7,
            1,
          ],
        }
      }
      values={
        MotionValuesMap {
          "hasMounted": true,
          "onUpdate": undefined,
          "output": [Function],
          "unsubscribers": Map {
            "scale" => [Function],
            "opacity" => [Function],
          },
          "values": Map {
            "scale" => MotionValue {
              "canTrackVelocity": true,
              "current": 0,
              "lastUpdated": 0,
              "notifySubscriber": [Function],
              "parent": undefined,
              "prev": undefined,
              "renderSubscribers": Set {
                [Function],
              },
              "scheduleVelocityCheck": [Function],
              "setChild": [Function],
              "stopAnimation": [Function],
              "timeDelta": 0,
              "transformer": undefined,
              "updateAndNotify": [Function],
              "updateSubscribers": Set {
                [Function],
              },
              "velocityCheck": [Function],
            },
            "opacity" => MotionValue {
              "canTrackVelocity": true,
              "current": 1,
              "lastUpdated": 0,
              "notifySubscriber": [Function],
              "parent": undefined,
              "prev": undefined,
              "renderSubscribers": Set {
                [Function],
              },
              "scheduleVelocityCheck": [Function],
              "setChild": [Function],
              "stopAnimation": [Function],
              "timeDelta": 0,
              "transformer": undefined,
              "updateAndNotify": [Function],
              "updateSubscribers": Set {
                [Function],
              },
              "velocityCheck": [Function],
            },
          },
        }
      }
    />
    <div
      className="points"
      style={
        Object {
          "opacity": 1,
          "transform": "scale(0) translateZ(0)",
        }
      }
    >
      +
      23
    </div>
  </ForwardRef(MotionComponent)>
</Points>
`;
